### PR TITLE
Feat/rotation offset

### PR DESCRIPTION
**Pull request by thestonefox**

### CODE REVIEW

File: Runtime/Data/Operation/TransformEulerRotationMutator.cs [hunk 1]
```diff
@@ -22,25 +34,87 @@ protected override float GetLocalAxisValue(int axis)
...
+        /// Returns the <see cref="Origin"/> position if a <see cref="Origin"/> is defined.
+        /// </summary>
+        /// <returns>The origin position.</returns>
+        protected virtual Vector3 GetOriginPosition()
+        {
+            return Origin != null ? Origin.transform.position : Vector3.zero;
+        }
+
+        /// <summary>
+        /// Applies the position of the <see cref="Origin"/> to the <see cref="Target"/> to ensure it rotates around the set origin.
```
Comment (bddckr) [inline Runtime/Data/Operation/TransformEulerRotationMutator.cs]:
```suggestion
        /// Applies the position of the <see cref="Origin"/> to the <see cref="TransformPropertyMutator.Target"/> to ensure it rotates around the set origin.
```

File: Runtime/Data/Operation/TransformPositionMutator.cs [hunk 2]
```diff
@@ -2,20 +2,20 @@
 {
     using UnityEngine;
     using Malimbe.MemberClearanceMethod;
-    using Malimbe.PropertySerializationAttribute;
     using Malimbe.XmlDocumentationAttribute;
+    using Malimbe.PropertySerializationAttribute;
```
Comment (bddckr) [inline Runtime/Data/Operation/TransformPositionMutator.cs]: Why was this changed? Now it's no longer in alphabetical order.

File: Runtime/Data/Type/Transformation/CollectionAggregator.cs [hunk 3]
```diff
@@ -23,6 +23,23 @@ public abstract class CollectionAggregator<TInput, TOutput, TEvent, TCollection,
         [field: DocumentedByXml]
         public TCollection Collection { get; set; }
 
+        /// <summary>
+        /// Processes the current collection and sums the collection total.
```
Comment (bddckr) [inline Runtime/Data/Type/Transformation/CollectionAggregator.cs]:
```suggestion
        /// Processes the current collection and aggregates the collection total.
```

File: Runtime/Data/Type/Transformation/CollectionAggregator.cs [hunk 4]
```diff
@@ -23,6 +23,23 @@ public abstract class CollectionAggregator<TInput, TOutput, TEvent, TCollection,
...
+        /// </summary>
+        /// <returns>The summed value of all values in the collection.</returns>
```
Comment (bddckr) [inline Runtime/Data/Type/Transformation/CollectionAggregator.cs]:
```suggestion
        /// <returns>The aggregated value of all values in the collection.</returns>
```

File: Runtime/Data/Type/Transformation/CollectionAggregator.cs [hunk 5]
```diff
@@ -23,6 +23,23 @@ public abstract class CollectionAggregator<TInput, TOutput, TEvent, TCollection,
...
+        public virtual TOutput Transform()
+        {
+            return Transform(Collection.NonSubscribableElements[Collection.CurrentIndex]);
+        }
+
+        /// <summary>
+        /// Processes the current collection and sums the collection total.
```
Comment (bddckr) [inline Runtime/Data/Type/Transformation/CollectionAggregator.cs]:
```suggestion
        /// Processes the current collection and aggregates the collection total.
```

File: Runtime/Data/Type/Transformation/CollectionAggregator.cs [hunk 6]
```diff
@@ -23,6 +23,23 @@ public abstract class CollectionAggregator<TInput, TOutput, TEvent, TCollection,
         [field: DocumentedByXml]
         public TCollection Collection { get; set; }
 
+        /// <summary>
+        /// Processes the current collection and sums the collection total.
+        /// </summary>
+        /// <returns>The summed value of all values in the collection.</returns>
+        public virtual TOutput Transform()
+        {
+            return Transform(Collection.NonSubscribableElements[Collection.CurrentIndex]);
```
Comment (bddckr) [inline Runtime/Data/Type/Transformation/CollectionAggregator.cs]: Just at a single index? Either this is wrong or the XML docs for this method (and `DoTransform`) should explain that this only operates on the `collection's current index` or something.

Comment (thestonefox) [reply Runtime/Data/Type/Transformation/CollectionAggregator.cs]: This is an implementation detail, it isn't required public knowledge about how it's performing the aggregation on the internal collection.

And the way it does that is to simply call `Transform(ObjectAtCurrentIndex)` 

which internally goes to the `Transform(Object)` method, which sets the given `Object` to the index of the collection at the `CurrentIndex`

So doing:

```
CurrentIndex = 1;
Transform(Object);

>>> does

Collection[CurrentIndex] = Object
ProcessCollection()
```

is the same as doing

```
CurrentIndex = 1
Transform()

>>> does

Transform()

Transform(ObjectAtCurrentIndex);
Collection[CurrentIndex] = Object
ProcessCollection()
```

File: Runtime/Data/Operation/TransformPositionMutator.cs (hunk 2 above)
Comment (thestonefox) [reply Runtime/Data/Operation/TransformPositionMutator.cs]: To match the style in most of the other scripts where namespaces flow with length to look neat. This should be resolved when we do a codebase wide operation applying a single rule of how we want the namespaces to work.

But for now, more often than not the namespaces aren't sorted alphabetically.
